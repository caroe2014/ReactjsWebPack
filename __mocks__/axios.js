
const mockAxios = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.create = () => mockAxios;
mockAxios.get = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.post = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.put = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.delete = jest.fn(() => Promise.resolve({data: {}}));
mockAxios.interceptors = {
  response: {
    use: jest.fn()
  },
  request: {
    use: jest.fn()
  }
};

export default mockAxios;
