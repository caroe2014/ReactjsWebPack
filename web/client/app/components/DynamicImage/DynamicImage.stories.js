// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import DynamicImage from './DynamicImageExample';

storiesOf('DynamicImage', module)
  .add('DynamicImage', () => {
    const siteItem = {
      image: 'http://eatstreet.co.id/wp-content/uploads/2016/11/cropped-MASTER-FINAL_LOGO-OVAL-EAT-STREET-1.png',
      name: 'Eat Street',
      availableAt: {
        opens: '9:00am',
        closes: '5:00pm'
      },
      address: ['1000, windward concourse, Sometown', 'Alpharetta,GA, USA - 500100']
    };
    return (
      <DynamicImage keyProps={siteItem} />
    );
  });
