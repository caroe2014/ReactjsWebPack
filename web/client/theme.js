// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
