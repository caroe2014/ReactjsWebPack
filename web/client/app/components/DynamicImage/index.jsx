// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

// global Image
import React, { Component } from 'react';
import config from 'app.config';
import axios from 'axios';

class DynamicImage extends Component {

  constructor (props) {
    super(props);
    this.fadeInTime = props.fadeInTime || 0.3;
    this.dynamicImageHd = null;
    this.srcPreload = `${config.webPaths.assets}default_image.png`;

    this.dynamicImageContainer = {
      position: 'relative',
      width: '100%',
      height: '100%',
      overflow: 'hidden',
      backgroundColor: '#eee'
    };

    this.dynamicImageLoaded = {
      position: 'absolute',
      zIndex: 2,
      top: 0,
      right: 0,
      bottom: 0,
      left: 0,
      transition: `opacity ${this.fadeInTime}s ease`,
      opacity: 0,
      height: '100%',
      width: '100%',
      backgroundRepeat: 'no-repeat',
      backgroundSize: this.props.bgSize ? this.props.bgSize : 'contain',
      backgroundPositionY: '50%',
      backgroundPositionX: '50%',
      border: 'solid 1px white',
      boxSizing: 'border-box'
    };
  }

  componentDidMount () {
    if (this.props.src) {
      axios.get(`${this.props.src}`,
        { responseType: 'blob' })
        .then((response) => {
          var reader = new window.FileReader();
          reader.readAsDataURL(response.data);
          /* istanbul ignore next */
          reader.onload = () => {
            const url = reader.result;
            if (this.dynamicImageHd) {
              this.dynamicImageHd.style.backgroundImage = `url('${url}')`;
              this.dynamicImageHd.style.opacity = 1;
              this.dynamicImageLow.style.opacity = 0;
              setTimeout(function () {
                this.dynamicImageLow.style.display = 'none';
              }.bind(this), 1000 * this.fadeInTime);
            }
          };
        });
    }
  };

  render () {
    return (
      <div className='dynamic-image-container' style={this.dynamicImageContainer}>
        <div
          className='dynamic-image'
          style={this.dynamicImageLoaded}
          alt=''
          ref={imageLoadedElem => (this.dynamicImageHd = imageLoadedElem)}
        />
        <div className='default-image' alt='' ref={imageLoadedElem => (this.dynamicImageLow = imageLoadedElem)} style={{
          backgroundImage: `url('${this.srcPreload}')`,
          position: 'absolute',
          zIndex: 1,
          top: 0,
          right: 0,
          bottom: 0,
          left: 0,
          opacity: 1,
          transition: `opacity ${this.fadeInTime}s ease`,
          backgroundRepeat: 'no-repeat',
          backgroundPosition: 'center',
          backgroundSize: 'contain',
          height: '100%',
          width: '100%',
          border: 'solid 1px #ccc',
          boxSizing: 'border-box'
        }}
        />
      </div>
    );
  }

}

export default DynamicImage;
