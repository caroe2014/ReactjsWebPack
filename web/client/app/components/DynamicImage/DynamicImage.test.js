import React from 'react';
import {
  mount,
  shallow,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import theme from 'web/client/theme';
import { ThemeProvider } from 'styled-components';
import axios from 'axios';

import DynamicImage from '.';
configure({
  adapter: new Adapter()
});

jest.mock('axios');

describe('DynamicImage', () => {

  it('does a snapshot test', () => {
    const wrapper = mount(<ThemeProvider theme={theme}><DynamicImage theme={theme} min={1}/></ThemeProvider>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    const getSpy = jest.spyOn(axios, 'get');
    expect(mount(<ThemeProvider theme={theme}><DynamicImage theme={theme} bgSize={'contain'} src='testURL' min={1}/></ThemeProvider>).exists()).toBe(true);
    expect(getSpy).toBeCalled();
  });

});
