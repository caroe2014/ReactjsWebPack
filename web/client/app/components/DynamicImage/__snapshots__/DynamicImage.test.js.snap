// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicImage does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <DynamicImage
    min={1}
    theme={
      Object {
        "colors": Object {
          "addOnTab": "#347CFF",
          "addToCart": "#347CFF",
          "addonBg": "#EEEEEE",
          "backgroundGrey": "#E6E6E6",
          "black": "#000000",
          "close": "red",
          "contextBackground": "#f0f0f3",
          "darkGrey": "#464646",
          "disableBackground": "#E4E4E4",
          "disableButton": "#D1D1D1",
          "error": "#cc0000",
          "footer": "#48a942",
          "glutenFree": "#8d7356",
          "healthy": "#359919",
          "hint": "#a6a6a6",
          "inputBottomBorder": "lightgray",
          "itemBorder": "lightgrey",
          "itemTile": "#3C474C0",
          "light": "white",
          "loader": "#ababab",
          "loginBackground": "#303030",
          "loginBoxColor": "#434149",
          "loginField": "#AAA7AC",
          "loginFormBackground": "#484848",
          "navigationMenu": "#48a942",
          "navigationMenuHover": "cornflowerblue",
          "new": "#ff6000",
          "open": "#48a942",
          "paymentSuccessTitle": "#505050",
          "placeHolder": "#909090",
          "primary": "#347CFF",
          "primaryButton": "#347CFF",
          "primaryText": "#383838",
          "secondaryText": "#a6a6a6",
          "spicy": "#ed1c24",
          "text": "black",
          "textGrey": "#505050",
          "tile": "#505050",
          "validationError": "#cc0000",
          "vegan": "#0d4a22",
          "vegetarian": "#0d4a22",
        },
        "fontFamily": "Roboto",
        "fontSize": Object {
          "lg": "1.8rem",
          "md": "1.3rem",
          "nm": "1rem",
          "sm": "0.75rem",
          "xl": "2.4rem",
        },
        "fontSizeFromProps": [Function],
        "mediaBreakpoints": Object {
          "desktop": [Function],
          "mobile": [Function],
          "tablet": [Function],
        },
        "nav": Object {
          "height": "70px",
        },
        "receiptFontFamily": "Roboto",
        "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
        "spacing": Object {
          "lg": "4em",
          "md": "2em",
          "nm": "1em",
          "none": "0em",
          "sm": "0.5em",
          "xl": "8em",
        },
      }
    }
  >
    <div
      className="dynamic-image-container"
      style={
        Object {
          "backgroundColor": "#eee",
          "height": "100%",
          "overflow": "hidden",
          "position": "relative",
          "width": "100%",
        }
      }
    >
      <div
        alt=""
        className="dynamic-image"
        style={
          Object {
            "backgroundPositionX": "50%",
            "backgroundPositionY": "50%",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "contain",
            "border": "solid 1px white",
            "bottom": 0,
            "boxSizing": "border-box",
            "height": "100%",
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transition": "opacity 0.3s ease",
            "width": "100%",
            "zIndex": 2,
          }
        }
      />
      <div
        alt=""
        className="default-image"
        style={
          Object {
            "backgroundImage": "url('application/ui/static/assets/default_image.png')",
            "backgroundPosition": "center",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "contain",
            "border": "solid 1px #ccc",
            "bottom": 0,
            "boxSizing": "border-box",
            "height": "100%",
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transition": "opacity 0.3s ease",
            "width": "100%",
            "zIndex": 1,
          }
        }
      />
    </div>
  </DynamicImage>
</ThemeProvider>
`;
